tests/integration/test_pyspark_savedmodel_artifact.py

```diff
@@ -0,0 +1,148 @@
+import time
+import urllib
+
+import pandas as pd
+import pytest
+from pyspark.ml.classification import LogisticRegression
+from pyspark.ml.feature import VectorAssembler
+from pyspark.sql import SparkSession
+
+import bentoml
+from bentoml.server.api_server import BentoAPIServer
+from tests.bento_service_examples.pyspark_classifier import PysparkClassifier
+
+
+def _wait_until_ready(_host, timeout, check_interval=0.5):
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        try:
+            if (
+                urllib.request.urlopen(f'http://{_host}/healthz', timeout=0.1).status
+                == 200
+            ):
+                break
+        except Exception:
+            time.sleep(check_interval - 0.1)
+        else:
+            raise AssertionError(f"server didn't get ready in {timeout} seconds")
+
+
+train_data = [[0, -1.0], [1, 1.0]]
+train_pddf = pd.DataFrame(train_data, columns=["label", "feature1"])
+test_data = [-5.0, 5.0, -0.5, 0.5]
+test_pddf = pd.DataFrame(test_data, columns=["feature1"])
+
+
+@pytest.fixture(scope="module")
+def spark_session():
+    return SparkSession.builder.appName("BentoService").getOrCreate()
+
+
+@pytest.fixture(scope="module")
+def pyspark_model(spark_session):
+    # Put pandas training df into Spark df form with Vector features
+    train_spdf = spark_session.createDataFrame(train_pddf)
+    assembler = VectorAssembler(inputCols=['feature1'], outputCol='features')
+    train_spdf = assembler.transform(train_spdf).select(['features', 'label'])
+
+    # Train model (should result in x=neg => y=0, x=pos => y=1)
+    lr = LogisticRegression()
+    lr_model = lr.fit(train_spdf)
+
+    return lr_model
+
+
+@pytest.fixture(scope="module")
+def pyspark_svc(pyspark_model):
+    svc = PysparkClassifier()
+    svc.pack('model', pyspark_model)
+
+    return svc
+
+
+@pytest.fixture(scope="module")
+def pyspark_svc_saved_dir(tmp_path_factory, pyspark_svc):
+    """Save a PySpark MLLib BentoService and return the saved directory."""
+    tmpdir = str(tmp_path_factory.mktemp("pyspark_svc"))
+    pyspark_svc.save_to_dir(tmpdir)
+
+    return tmpdir
+
+
+@pytest.fixture()
+def pyspark_svc_loaded(pyspark_svc_saved_dir):
+    """Return a PySpark BentoService that has been saved and loaded."""
+    return bentoml.load(pyspark_svc_saved_dir)
+
+
+@pytest.fixture()
+def pyspark_image(pyspark_svc_saved_dir):
```

Comment:
could we use `build_api_server_docker_image` in `tests/integration/utils.py` here?